muikku-core-plugins/src/main/java/fi/otavanopisto/muikku/plugins/user/rest/UserPluginRESTService.java

```diff
@@ -66,6 +67,8 @@ public Response updateCredentials(UserCredentials userCredentialChange) {
       return Response.noContent().build();
     } catch (SchoolDataBridgeUnauthorizedException e) {
       return Response.status(Status.FORBIDDEN).build();
+    } catch (SchoolDataBridgeException e) {
+      return Response.status(Status.BAD_REQUEST).build();
```

Comment:
Since SchoolDataBridgeException contains a (HTTP) statusCode, this catch block and the one above could actually be combined into one:

```
catch (SchoolDataBridgeException e) {
  return Response.status(e.getStatusCode()).build();
}
```

In addition to that, there are two places in `PyramusRestClient` that throw `SchoolDataBridgeInternalException`. If both of them are rewritten as...

`throw new SchoolDataBridgeException(response.getStatus(), String.format("Received http error %d (%s) when requesting %s", response.getStatus(), response.getEntity(), path));`

...then any HTTP error code from any REST call to Pyramus can be passed along in `SchoolDataBridgeException`.